src/core/lib/support/closure_ref.h

```diff
@@ -0,0 +1,248 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_SUPPORT_CLOSURE_REF_H
+#define GRPC_CORE_LIB_SUPPORT_CLOSURE_REF_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <utility>
+
+namespace grpc_core {
+
+// Value type reference to some closure
+// Template arguments list argument types to the closure
+// Closures have an implicit scheduling policy bound when they are created
+template <typename... Args>
+class ClosureRef {
```

Comment:
If this should only actually get made through MakeClosure or a copy constructor, consider making the regular (non-copy) constructors private and having MakeClosure be a friend - this isn't API but it'll make things less confusing for people trying to add to core in the future.

Reply:
Done

Reply:
Maybe not pushed? I don't see it.

Reply:
pushed